#!/usr/bin/env python
PACKAGE = "iarc7_vision"

from dynamic_reconfigure.parameter_generator_catkin import *
import math

gen = ParameterGenerator()

gen.add('ght_pos_thresh',        int_t,    0, '', 0, 0, 1000)
gen.add('ght_angle_thresh',      int_t,    0, '', 0, 0, 1000)
gen.add('ght_scale_thresh',      int_t,    0, '', 0, 0, 1000)
gen.add('ght_canny_low_thresh',  int_t,    0, '', 0, 0, 1000)
gen.add('ght_canny_high_thresh', int_t,    0, '', 0, 0, 1000)
gen.add('ght_dp',                double_t, 0, '', 0, 0, 10)
gen.add('ght_levels',            int_t,    0, '', 0, 0, 1000)
gen.add('ght_angle_step',        double_t, 0, '', 0, 0, 10)
gen.add('ght_scale_step',        double_t, 0, '', 0, 0, 1)

gen.add('hsv_slice_h_green_min', int_t, 0, '', 0, 0, 180)
gen.add('hsv_slice_h_green_max', int_t, 0, '', 0, 0, 180)
gen.add('hsv_slice_h_red1_min',  int_t, 0, '', 0, 0, 180)
gen.add('hsv_slice_h_red1_max',  int_t, 0, '', 0, 0, 180)
gen.add('hsv_slice_h_red2_min',  int_t, 0, '', 0, 0, 180)
gen.add('hsv_slice_h_red2_max',  int_t, 0, '', 0, 0, 180)
gen.add('hsv_slice_s_min',       int_t, 0, '', 0, 0, 255)
gen.add('hsv_slice_s_max',       int_t, 0, '', 0, 0, 255)
gen.add('hsv_slice_v_min',       int_t, 0, '', 0, 0, 255)
gen.add('hsv_slice_v_max',       int_t, 0, '', 0, 0, 255)

gen.add('morphology_size', int_t, 0, '', 3, 3, 9)
gen.add('morphology_iterations', int_t, 0, '', 1, 1, 10)

exit(gen.generate(PACKAGE, "iarc7_vision", "RoombaEstimator"))
